widgetssdk/src/main/java/com/glia/widgets/core/dialog/model/DialogState.kt

```diff
@@ -0,0 +1,23 @@
+package com.glia.widgets.core.dialog.model
+
+import com.glia.androidsdk.comms.MediaUpgradeOffer
+
+internal sealed interface DialogState {
+    object None : DialogState
+    object OverlayPermission : DialogState
+    object ExitQueue : DialogState
+    data class StartScreenSharing(val operatorName: String?) : DialogState
```

Comment:
What if you were to change the DialogState into an internal class that would have operatorName, mediaUpgradeOffer, mode and DialogType as arguments?

You would probably have to create custom contstructors for the argumented types, but this way the different types would be equivalent and the arguments would be there for the types that need them

Reply:
After implementing the proposed solution using generics, I realized that it did not provide any benefits, so I decided to leave everything as it was. Let me explain in detail.

```kotlin
sealed interface Result<out T> {
    object Loading: Result<Nothing>
    data class Error(val ex: Exception): Result<Nothing>
    data class Success<out T>(val data: T): Result<T>
}
``` 
In the Result<T> interface, T can represent any type of value that might be returned upon the success of some operation. This allows us to keep the type safety by knowing that Result<User> will hold a User in case of success, Result<Post> will hold Post, and so on.

Now, let's compare this to the DialogState interface. 

In DialogState, each state is explicitly defined as an object or a data class with specifically defined parameters. There's no need for a generic type because each possible state of the dialog is explicitly enumerated in the interface. 

For example, StartScreenSharing holds an `operatorName` of type String and this would never be a different type (like User, Post, or Int). Similarly, MediaUpgrade holds a mediaUpgradeOffer of type MediaUpgradeOffer, and so on. 

Essentially, we don't use a generic type in DialogState because we don't have a situation where we might have a DialogState<User>, DialogState<Post>, DialogState<Int>, etc. Each state is distinct and holds specifically defined data related to that state.

Reply:
But during that local update, I found that we had set a lot of `DialogInterface.OnCancelListener` to the dialogs that are not cancelable.
Will provide that update soon.